src/rules/selector-type-case/index.js

```diff
@@ -0,0 +1,66 @@
+import selectorParser from "postcss-selector-parser"
+import {
+  cssRuleHasSelectorEndingWithColon,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "selector-type-case"
+
+export const messages = ruleMessages(ruleName, {
+  expected: (h, v) => `Expected "${h}" to be "${v}"`,
+})
+
+export default function (expectation) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [
+        "lower",
+        "upper",
+      ],
+    })
+    if (!validOptions) { return }
+
+    root.walkRules(rule => {
+      // Ignore keyframe selectors
+      if (rule.parent.type === "atrule" && rule.parent.name === "keyframes") {
+        return
+      }
+
+      if (cssRuleHasSelectorEndingWithColon(rule)) { return }
+
+      selectorParser(selectorAST => {
+        selectorAST.eachTag(tag => {
```

Comment:
You could probably deconstruct tag here for cleanliness e.g. 

const { parent, sourceIndex, value } = tag